### PR TITLE
MOSIP-31715,MOSIP-31940,MOSIP-31570

### PR DESCRIPTION
**Pull request by dhanendra06**

### CODE REVIEW

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/constant/SchemaErrorCode.java [hunk 1]
```diff
@@ -32,6 +32,9 @@ public enum SchemaErrorCode {
 	SUB_TYPE_REQUIRED_EXCEPTION("KER-SCH-017", "SubType is required for field %s"),
 	BIO_ATTRIBUTES_REQUIRED_EXCEPTION("KER-SCH-018", "BioAttributes are required for field %s"),
 	BIO_ATTRIBUTES_DUPLICATED_EXCEPTION("KER-SCH-019", "Same BioAttributes used in field with same SubType : %s"),
+
+	PUBLISHED_SCHEMA_EXCEPTION("KER-SCH-022", "Published identity schema can't be deleted"),
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/constant/SchemaErrorCode.java:36]: @dhanendra06  why is the error code jumped from 019 to 022?

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java [hunk 2]
```diff
@@ -306,9 +306,8 @@ public void testPublishIdentitySchemaWithException() throws Exception {
 	
 	@Test
 	@WithUserDetails("global-admin")
-	public void testDeleteIdentitySchema() throws Exception {		
-		Mockito.when(identitySchemaRepository.deleteIdentitySchema(Mockito.anyString(),  Mockito.any(LocalDateTime.class), 
-				Mockito.anyString())).thenReturn(1);		
+	public void testDeleteIdentitySchema() throws Exception {
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java:309]: Test method naming convention is not followed.

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java [hunk 3]
```diff
@@ -306,9 +306,8 @@ public void testPublishIdentitySchemaWithException() throws Exception {
...
+		Mockito.when(identitySchemaRepository.findIdentitySchemaById(Mockito.anyString())).thenReturn(draftSchema);
 		identitySchemaService.deleteSchema("test-test");
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java:311]: There is no assertion nor exception that is caught in the test method.

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java [hunk 4]
```diff
@@ -322,9 +321,17 @@ public void testDeleteIdentitySchemaFailed() throws Exception {
 	
 	@Test(expected = MasterDataServiceException.class)
 	@WithUserDetails("global-admin")
-	public void testDeleteIdentitySchemaFailedUpdate() throws Exception {
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java:325]: Test method naming convention is not followed.
methodName_testCase_testResult

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java [hunk 5]
```diff
@@ -322,9 +321,17 @@ public void testDeleteIdentitySchemaFailed() throws Exception {
...
-				Mockito.anyString())).thenThrow(DataAccessLayerException.class);		
+	public void testDeleteIdentitySchemaFailedUpdate() throws Exception {
+		Mockito.when(identitySchemaRepository.findIdentitySchemaById(
+				Mockito.anyString())).thenThrow(DataAccessLayerException.class);
+		identitySchemaService.deleteSchema("test-test");
+	}
+
+	@Test(expected = RequestException.class)
+	@WithUserDetails("global-admin")
+	public void testDeletePublishedIdentitySchema() throws Exception {
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java:332]: Test method naming convention is not followed.
methodName_testCase_testResult

File: db_scripts/mosip_master/dml.sql [hunk 6]
```diff
@@ -104,6 +104,9 @@ TRUNCATE TABLE master.template cascade ;
 
 \COPY master.template (id,name,descr,file_format_code,model,file_txt,module_id,module_name,template_typ_code,lang_code,is_active,cr_by,cr_dtimes) FROM './dml/master-template.csv' delimiter ',' HEADER  csv;
 
+\COPY master.blocklisted_words (word,descr,lang_code,is_active,cr_by,cr_dtimes) FROM './dml/master-blacklisted_words.csv' delimiter ',' HEADER  csv;
```
Comment (ase-101) [inline db_scripts/mosip_master/dml.sql:107]: Blocklisted words are country-specific data it should not be part of DML scripts.

**Review by ase-101 — CHANGES_REQUESTED** [review]

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java (hunk 2 above)
Comment (dhanendra06) [reply admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java]: taken care

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java (hunk 3 above)
Comment (dhanendra06) [reply admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java]: taken care

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java (hunk 4 above)
Comment (dhanendra06) [reply admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java:325]: taken care

File: admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java (hunk 5 above)
Comment (dhanendra06) [reply admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/SchemaServiceTest.java]: taken care

File: db_scripts/mosip_master/dml.sql (hunk 6 above)
Comment (dhanendra06) [reply db_scripts/mosip_master/dml.sql]: removed and as we have already added in mosip-data a part of merging back to develop so pr is not required for mosip-data

**Review by dhanendra06 — COMMENTED** [review]
local build is passing but travis 2 test cases are failing for all the pr.

Error: RegistrationCenterControllerTest.t017deleteRegistrationCenterTest:380 expected: but was:
Error: RegistrationCenterControllerTest.t023decommissionRegCenterFailTest1:522 expected:<KER-MSD-[351]> but was:<KER-MSD-[215]>

**Review by ase-101 — APPROVED** [review]

**Pull request merged**